src/os/bluestore/BlueStore.cc

```diff
@@ -11650,7 +11643,7 @@ void BlueStore::_kv_sync_thread()
 
 	} else {
 	  ceph_assert(txc->state == TransContext::STATE_KV_SUBMITTED);
-	  txc->log_state_latency(logger, l_bluestore_state_kv_queued_lat);
+	  ceph_assert(0 == "should be configured to be impossible");
```

Comment:
nit, why bother checking `ceph_assert(txc->state == TransContext::STATE_KV_SUBMITTED);` if all transactions come from `kv_queue_unsubmitted`.

Reply:
Yeah, I messed this up in one of the rebases, I'll fix it.

Reply:
That loop is actually over kv_committing which comes from kv_queue.  My assert there is a mistake, I added it during debugging to confirm that bluestore was doing what I thought it was doing and failed subsequently to remove it.  That path (and the first assert) reflect the bluestore_sync_submit_transaction path in _txc_state_proc by which the transaction will be put into the kv_queue (though not kv_queue_unsubmitted) *after* being synchronously submitted to the kv backend.  kv_queue_unsubmitted appears to be a subset of kv_queue consisting of those kv operations which have not yet been submitted.

Reply:
Actually, looking at it again, l_bluestore_state_kv_queued_lat seems to be intended to track removal from the queue, not application.  I'm going to move that back to here from _txc_apply_kv.